src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -479,14 +479,14 @@ JRT_LEAF(jshort, SharedRuntime::f2hf(jfloat  x))
     exp = -15;
     msb = 0x00800000;
   }
-  jint f_signif_bits = doppel & 0x007fffff | msb;
+  jint f_signif_bits = ((doppel & 0x007fffff) | msb);
 
   // Significand bits as if using rounding to zero
   jshort signif_bits = (jshort)(f_signif_bits >> (13 + exp_delta));
 
-  jint lsb = f_signif_bits & (1 << 13 + exp_delta);
-  jint round  = f_signif_bits & (1 << 12 + exp_delta);
-  jint sticky = f_signif_bits & ((1 << 12 + exp_delta) - 1);
+  jint lsb = (f_signif_bits & (1 << 13 + exp_delta));
```

Comment:
It will be more clear to have the parenthesis like this:
jint lsb = f_signif_bits & (1 << (13 + exp_delta));